src/sst/core/event.cc

```diff
@@ -24,10 +24,68 @@ const SST::Event::id_type SST::Event::NO_ID = std::make_pair(0, -1);
 
 Event::~Event() {}
 
+extern std::map<std::string, uint64_t> eventHandlers;
+extern std::map<std::string, uint64_t> eventSendCounters;
+extern std::map<std::string, uint64_t> eventRecvCounters;
 
 void Event::execute(void)
 {
+
+    #ifdef EVENT_PROFILING
+    #ifdef HIGH_RESOLUTION_CLOCK
+    auto eventStart = std::chrono::high_resolution_clock::now();
+    #else
+    struct timeval eventStart, eventEnd, eventDiff;
+    gettimeofday(&eventStart, NULL);
+    #endif
+    #endif
+
     delivery_link->deliverEvent(this);
+
+    #ifdef EVENT_PROFILING
+    #ifdef HIGH_RESOLUTION_CLOCK
+    auto eventFinish = std::chrono::high_resolution_clock::now();
+    #else
+    gettimeofday(&eventEnd, NULL);
+    timersub(&eventEnd, &eventStart, &eventDiff);
+    #endif
+
+    // Track receiver processing time
+    auto eventHandler = eventHandlers.find(getLastComponentName());
+    if(eventHandler != eventHandlers.end())
+    {
+        #ifdef HIGH_RESOLUTION_CLOCK
+        eventHandler->second += std::chrono::duration_cast<std::chrono::nanoseconds>(eventFinish - eventStart).count();
+        #else
+        eventHandler->second += eventDiff.tv_usec + eventDiff.tv_sec * 1e6;
+        #endif
+    }
+
+    // Track sending and receiving counters
+    auto eventCount = eventRecvCounters.find(getLastComponentName());
+    if(eventCount != eventRecvCounters.end())
+    {
+        eventCount->second ++;
+    }else{
+        if(getLastComponentName() != "")
+        {
+            eventRecvCounters.insert(std::pair<std::string, uint64_t>(getLastComponentName().c_str(), 1));
+            eventHandlers.insert(std::pair<std::string, uint64_t>(getLastComponentName().c_str(), 0));
+        }
+    }
+    auto eventSend = eventSendCounters.find(getFirstComponentName());
+    if(eventSend != eventSendCounters.end())
+    {
+       eventSend->second++;
+    }else{
+        // Insert handler and counter for the subcomponent so that all link traffic is monitored
+        if(getFirstComponentName() != "")
+        {
+            eventSendCounters.insert(std::pair<std::string, uint64_t>(getFirstComponentName().c_str(),1));
```

Comment:
This has the possibility of concurrent accesses to the eventSendCounters object since it can happen for the first time after the run loop has started.